src/main/java/com/gorlah/kappabot/function/response/ImmutableBotResponse.java

```diff
@@ -0,0 +1,10 @@
+package com.gorlah.kappabot.function.response;
+
+import lombok.Value;
+
+@Value(staticConstructor = "of")
+public class ImmutableBotResponse implements BotResponse {
```

Comment:
What about a few additional static constructors on here to specify content type? E.g. ofTextMarkdown(String message).

I know it would dirty up this class a bit but it might make the calling code a bit cleaner and less error prone.

Reply:
I considered exactly that when I was working on this initially but never went back and did it. I'll think on it a bit more.

Reply:
I've added static factories in `BotResponses`.